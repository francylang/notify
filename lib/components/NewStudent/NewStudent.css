.new-student {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: #fff;
  height: 340px;
  width: 400px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 10px #bdbdbd; }
