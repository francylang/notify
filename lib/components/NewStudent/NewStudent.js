import React from 'react';
import '../../index.css';
import { Link } from 'react-router-dom';


const NewStudent = (props) => {
  const studentId = props.students.length

  let studentNameInput;
  let parentContactInput;
  let phoneInput;

  const handleSubmit = (event) => {
    event.preventDefault()
    props.addStudent({id: studentId,
                      name: studentNameInput.value,
                      parentContact: parentContactInput.value,
                      phone: phoneInput.value})
  }
  return(
    <div className='new-student'>
      <h2 className='create-new-student'>Create New Student</h2>
      <form className='add-new-student-form' onSubmit={handleSubmit}>
        <input ref={name => studentNameInput = name} type='text' placeholder='Student Name'/>
        <input ref={parent => parentContactInput = parent} type='text' placeholder='Parent Contact'/>
        <input ref={phone => phoneInput = phone} type='text' placeholder='phone'/>
        <input className='form-button' type='submit'/>
      </form>
    </div>
  )
}

export default NewStudent;
