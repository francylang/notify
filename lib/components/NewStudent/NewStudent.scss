$main-color: #1a90bf ;
$main-color-hover: #405780;


.new-student {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: #fff;
  height: 340px;
  width: 400px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 10px #bdbdbd;
}

.create-new-student {
  text-align: center;
  color: $main-color;

}

.add-new-student-form input {
  margin-top: 30px;
}

input {
  width: 100%;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #bdbdbd;
  font-size: 20px;
  font-weight: lighter;
  margin: 20px 0;
}
input:focus {
  outline: none;
  border-bottom: 2px solid $main-color;
}

.form-button {
  background-color: $main-color;
  border: none;
  border-radius: 10px;
  color: #fff;
  padding: 8px 20px;
  font-size: 16px;
  margin: 20px 0;
  width: 100%;
  cursor: pointer;
}

.form-button:hover {
  background-color: $main-color-hover;
}

.form-button:disabled {
  background-color: #bdbdbd;
}

@media screen and (max-width: 500px) {
  .new-student {
    width: 260px;
    margin-top: 100px;
  }
}
