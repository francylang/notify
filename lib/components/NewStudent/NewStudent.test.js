import React from 'react';
import NewStudent from './NewStudent.js';
import { shallow } from 'enzyme';

describe('<NewStudent />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<NewStudent students={[{ id: 5, name: 'Francy Lang', parentContact: 'Pam Lang', phone: '5029052136'}]}/>);
  });

  it('should exist', () => {

    expect(wrapper).toBeDefined();
  });

  it('should match the NewStudent snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});
