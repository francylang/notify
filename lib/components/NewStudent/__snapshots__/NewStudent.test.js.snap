// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewStudent /> should match the NewStudent snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="new-student"
>
    <h2
        className="create-new-student"
    >
        Create New Student
    </h2>
    <form
        className="add-new-student-form"
    >
        <input
            onChange={[Function]}
            placeholder="Student Name"
            type="text"
            value=""
        />
        <input
            onChange={[Function]}
            placeholder="Parent Contact"
            type="text"
            value=""
        />
        <input
            onChange={[Function]}
            placeholder="phone"
            type="text"
            value=""
        />
        <input
            className="form-button"
            disabled={true}
            onClick={[Function]}
            type="submit"
        />
    </form>
</div>,
  "nodes": Array [
    <div
      className="new-student"
>
      <h2
            className="create-new-student"
      >
            Create New Student
      </h2>
      <form
            className="add-new-student-form"
      >
            <input
                  onChange={[Function]}
                  placeholder="Student Name"
                  type="text"
                  value=""
            />
            <input
                  onChange={[Function]}
                  placeholder="Parent Contact"
                  type="text"
                  value=""
            />
            <input
                  onChange={[Function]}
                  placeholder="phone"
                  type="text"
                  value=""
            />
            <input
                  className="form-button"
                  disabled={true}
                  onClick={[Function]}
                  type="submit"
            />
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NewStudent
        students={
                Array [
                        Object {
                          "id": 5,
                          "name": "Francy Lang",
                          "parentContact": "Pam Lang",
                          "phone": "5029052136",
                        },
                      ]
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NewStudent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleSubmit": [Function],
        "props": Object {
          "students": Array [
            Object {
              "id": 5,
              "name": "Francy Lang",
              "parentContact": "Pam Lang",
              "phone": "5029052136",
            },
          ],
        },
        "refs": Object {},
        "state": Object {
          "id": 1,
          "name": "",
          "parentContact": "",
          "phone": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="new-student"
>
          <h2
                    className="create-new-student"
          >
                    Create New Student
          </h2>
          <form
                    className="add-new-student-form"
          >
                    <input
                              onChange={[Function]}
                              placeholder="Student Name"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="Parent Contact"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="phone"
                              type="text"
                              value=""
                    />
                    <input
                              className="form-button"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    />
          </form>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="new-student"
>
          <h2
                    className="create-new-student"
          >
                    Create New Student
          </h2>
          <form
                    className="add-new-student-form"
          >
                    <input
                              onChange={[Function]}
                              placeholder="Student Name"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="Parent Contact"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="phone"
                              type="text"
                              value=""
                    />
                    <input
                              className="form-button"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    />
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NewStudent
    students={
        Array [
            Object {
              "id": 5,
              "name": "Francy Lang",
              "parentContact": "Pam Lang",
              "phone": "5029052136",
            },
          ]
    }
/>,
}
`;
